I don't get the compilation error on GTK. Maybe we just need to trigger a clean build??

```
/app/webkit/WebKitBuild/Release/DerivedSources/WebKit/GeneratedSerializers.cpp:8455:88: error: template argument 1 is invalid
 8455 |     static_assert(std::is_same_v<std::remove_cvref_t<decltype(instance.contentInsets())>, WebCore::FloatBoxExtent>);
      |                                                                                        ^
/app/webkit/WebKitBuild/Release/DerivedSources/WebKit/GeneratedSerializers.cpp:8455:24: error: template argument 1 is invalid
 8455 |     static_assert(std::is_same_v<std::remove_cvref_t<decltype(instance.contentInsets())>, WebCore::FloatBoxExtent>);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/app/webkit/WebKitBuild/Release/DerivedSources/WebKit/GeneratedSerializers.cpp:8456:63: error: invalid use of incomplete type ‘const class WebKit::VisibleContentRectUpdateInfo’
 8456 |     static_assert(std::is_same_v<std::remove_cvref_t<decltype(instance.unobscuredRectInScrollViewCoordinates())>, WebCore::FloatRect>);
      |                                                               ^~~~~~~~
/app/webkit/WebKitBuild/Release/DerivedSources/WebKit/GeneratedSerializers.h:681:26: note: forward declaration of ‘class WebKit::VisibleContentRectUpdateInfo’
  681 | namespace WebKit { class VisibleContentRectUpdateInfo; }
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/app/webkit/WebKitBuild/Release/DerivedSources/WebKit/GeneratedSerializers.cpp:8456:63: error: invalid use of incomplete type ‘const class WebKit::VisibleContentRectUpdateInfo’
 8456 |     static_assert(std::is_same_v<std::remove_cvref_t<decltype(instance.unobscuredRectInScrollViewCoordinates())>, WebCore::FloatRect>);
      |                                                               ^~~~~~~~
      
```

--- a/Source/WebKit/CMakeLists.txt
+++ b/Source/WebKit/CMakeLists.txt
@@ -400,6 +400,7 @@ set(WebKit_SERIALIZATION_IN_FILES
     Shared/TextFlags.serialization.in
     Shared/TextRecognitionResult.serialization.in
     Shared/UpdateInfo.serialization.in
+    Shared/VisibleContentRectUpdateInfo.serialization.in
     Shared/WTFArgumentCoders.serialization.in
     Shared/WebCoreArgumentCoders.serialization.in
     Shared/WebEvent.serialization.in

--- a/Source/WebKit/DerivedSources-input.xcfilelist
+++ b/Source/WebKit/DerivedSources-input.xcfilelist
@@ -242,6 +242,7 @@ $(PROJECT_DIR)/Shared/TextFlags.serialization.in
 $(PROJECT_DIR)/Shared/TextRecognitionResult.serialization.in
 $(PROJECT_DIR)/Shared/UpdateInfo.serialization.in
 $(PROJECT_DIR)/Shared/UserInterfaceIdiom.serialization.in
+$(PROJECT_DIR)/Shared/VisibleContentRectUpdateInfo.serialization.in
 $(PROJECT_DIR)/Shared/WTFArgumentCoders.serialization.in
 $(PROJECT_DIR)/Shared/WebConnection.messages.in
 $(PROJECT_DIR)/Shared/WebCoreArgumentCoders.serialization.in

--- a/Source/WebKit/DerivedSources.make
+++ b/Source/WebKit/DerivedSources.make
@@ -546,6 +546,7 @@ SERIALIZATION_DESCRIPTION_FILES = \
 	Shared/TextFlags.serialization.in \
 	Shared/TextRecognitionResult.serialization.in \
 	Shared/UserInterfaceIdiom.serialization.in \
+	Shared/VisibleContentRectUpdateInfo.serialization.in \
 	Shared/WTFArgumentCoders.serialization.in \
 	Shared/WebCoreArgumentCoders.serialization.in \
 	Shared/WebEvent.serialization.in \

--- a/Source/WebKit/Shared/VisibleContentRectUpdateInfo.cpp
+++ b/Source/WebKit/Shared/VisibleContentRectUpdateInfo.cpp
@@ -36,61 +36,6 @@
 namespace WebKit {
 using namespace WebCore;
 
-void VisibleContentRectUpdateInfo::encode(IPC::Encoder& encoder) const
-{
-    encoder << m_exposedContentRect;
-    encoder << m_unobscuredContentRect;
-    encoder << m_contentInsets;
-    encoder << m_unobscuredContentRectRespectingInputViewBounds;
-    encoder << m_unobscuredRectInScrollViewCoordinates;
-    encoder << m_layoutViewportRect;
-    encoder << m_obscuredInsets;
-    encoder << m_unobscuredSafeAreaInsets;
-    encoder << m_scrollVelocity;
-    encoder << m_lastLayerTreeTransactionID;
-    encoder << m_scale;
-    encoder << m_viewStability;
-    encoder << m_isFirstUpdateForNewViewSize;
-    encoder << m_allowShrinkToFit;
-    encoder << m_enclosedInScrollableAncestorView;
-}
-
-bool VisibleContentRectUpdateInfo::decode(IPC::Decoder& decoder, VisibleContentRectUpdateInfo& result)
-{
-    if (!decoder.decode(result.m_exposedContentRect))
-        return false;
-    if (!decoder.decode(result.m_unobscuredContentRect))
-        return false;
-    if (!decoder.decode(result.m_contentInsets))
-        return false;
-    if (!decoder.decode(result.m_unobscuredContentRectRespectingInputViewBounds))
-        return false;
-    if (!decoder.decode(result.m_unobscuredRectInScrollViewCoordinates))
-        return false;
-    if (!decoder.decode(result.m_layoutViewportRect))
-        return false;
-    if (!decoder.decode(result.m_obscuredInsets))
-        return false;
-    if (!decoder.decode(result.m_unobscuredSafeAreaInsets))
-        return false;
-    if (!decoder.decode(result.m_scrollVelocity))
-        return false;
-    if (!decoder.decode(result.m_lastLayerTreeTransactionID))
-        return false;
-    if (!decoder.decode(result.m_scale))
-        return false;
-    if (!decoder.decode(result.m_viewStability))
-        return false;
-    if (!decoder.decode(result.m_isFirstUpdateForNewViewSize))
-        return false;
-    if (!decoder.decode(result.m_allowShrinkToFit))
-        return false;
-    if (!decoder.decode(result.m_enclosedInScrollableAncestorView))
-        return false;
-
-    return true;
-}
-
 String VisibleContentRectUpdateInfo::dump() const
 {
     TextStream stream;

--- a/Source/WebKit/Shared/VisibleContentRectUpdateInfo.h
+++ b/Source/WebKit/Shared/VisibleContentRectUpdateInfo.h
@@ -58,7 +58,10 @@ class VisibleContentRectUpdateInfo {
 public:
     VisibleContentRectUpdateInfo() = default;
 
-    VisibleContentRectUpdateInfo(const WebCore::FloatRect& exposedContentRect, const WebCore::FloatRect& unobscuredContentRect, const WebCore::FloatBoxExtent& contentInsets, const WebCore::FloatRect& unobscuredRectInScrollViewCoordinates, const WebCore::FloatRect& unobscuredContentRectRespectingInputViewBounds, const WebCore::FloatRect& layoutViewportRect, const WebCore::FloatBoxExtent& obscuredInsets, const WebCore::FloatBoxExtent& unobscuredSafeAreaInsets, double scale, OptionSet<ViewStabilityFlag> viewStability, bool isFirstUpdateForNewViewSize, bool allowShrinkToFit, bool enclosedInScrollableAncestorView, const WebCore::VelocityData& scrollVelocity, TransactionID lastLayerTreeTransactionId)
+    VisibleContentRectUpdateInfo(const WebCore::FloatRect& exposedContentRect, const WebCore::FloatRect& unobscuredContentRect, const WebCore::FloatBoxExtent& contentInsets,
+        const WebCore::FloatRect& unobscuredRectInScrollViewCoordinates, const WebCore::FloatRect& unobscuredContentRectRespectingInputViewBounds, const WebCore::FloatRect& layoutViewportRect,
+        const WebCore::FloatBoxExtent& obscuredInsets, const WebCore::FloatBoxExtent& unobscuredSafeAreaInsets, double scale, OptionSet<ViewStabilityFlag> viewStability,
+        bool isFirstUpdateForNewViewSize, bool allowShrinkToFit, bool enclosedInScrollableAncestorView, const WebCore::VelocityData& scrollVelocity, TransactionID lastLayerTreeTransactionId)
         : m_exposedContentRect(exposedContentRect)
         , m_unobscuredContentRect(unobscuredContentRect)
         , m_contentInsets(contentInsets)
@@ -83,7 +86,6 @@ public:
     const WebCore::FloatRect& unobscuredRectInScrollViewCoordinates() const { return m_unobscuredRectInScrollViewCoordinates; }
     const WebCore::FloatRect& unobscuredContentRectRespectingInputViewBounds() const { return m_unobscuredContentRectRespectingInputViewBounds; }
     const WebCore::FloatRect& layoutViewportRect() const { return m_layoutViewportRect; }
-    const WebCore::VelocityData& scrollVelocity() const { return m_scrollVelocity; }
     const WebCore::FloatBoxExtent& obscuredInsets() const { return m_obscuredInsets; }
     const WebCore::FloatBoxExtent& unobscuredSafeAreaInsets() const { return m_unobscuredSafeAreaInsets; }
 
@@ -93,12 +95,10 @@ public:
     bool isFirstUpdateForNewViewSize() const { return m_isFirstUpdateForNewViewSize; }
     bool allowShrinkToFit() const { return m_allowShrinkToFit; }
     bool enclosedInScrollableAncestorView() const { return m_enclosedInScrollableAncestorView; }
+    const WebCore::VelocityData& scrollVelocity() const { return m_scrollVelocity; }
     TransactionID lastLayerTreeTransactionID() const { return m_lastLayerTreeTransactionID; }
 
     MonotonicTime timestamp() const { return m_scrollVelocity.lastUpdateTime; }
-
-    void encode(IPC::Encoder&) const;
-    static WARN_UNUSED_RETURN bool decode(IPC::Decoder&, VisibleContentRectUpdateInfo&);
 
     String dump() const;
 
@@ -142,20 +142,5 @@ WTF::TextStream& operator<<(WTF::TextStream&, ViewStabilityFlag);
 WTF::TextStream& operator<<(WTF::TextStream&, const VisibleContentRectUpdateInfo&);
 
 } // namespace WebKit
-
-namespace WTF {
-
-template<> struct EnumTraits<WebKit::ViewStabilityFlag> {
-    using values = EnumValues<
-        WebKit::ViewStabilityFlag,
-        WebKit::ViewStabilityFlag::ScrollViewInteracting,
-        WebKit::ViewStabilityFlag::ScrollViewAnimatedScrollOrZoom,
-        WebKit::ViewStabilityFlag::ScrollViewRubberBanding,
-        WebKit::ViewStabilityFlag::ChangingObscuredInsetsInteractively,
-        WebKit::ViewStabilityFlag::UnstableForTesting
-    >;
-};
-
-} // namespace WTF
 
 #endif // ENABLE(UI_SIDE_COMPOSITING)

--- a/Source/WebKit/Shared/VisibleContentRectUpdateInfo.serialization.in
+++ b/Source/WebKit/Shared/VisibleContentRectUpdateInfo.serialization.in
@@ -1,0 +1,52 @@
+# Copyright (C) 2023 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if ENABLE(UI_SIDE_COMPOSITING)
+
+[OptionSet] enum class WebKit::ViewStabilityFlag : uint8_t {
+    ScrollViewInteracting,
+    ScrollViewAnimatedScrollOrZoom,
+    ScrollViewRubberBanding,
+    ChangingObscuredInsetsInteractively,
+    UnstableForTesting,
+};
+
+class WebKit::VisibleContentRectUpdateInfo {
+    WebCore::FloatRect exposedContentRect();
+    WebCore::FloatRect unobscuredContentRect();
+    WebCore::FloatBoxExtent contentInsets();
+    WebCore::FloatRect unobscuredRectInScrollViewCoordinates();
+    WebCore::FloatRect unobscuredContentRectRespectingInputViewBounds();
+    WebCore::FloatRect layoutViewportRect();
+    WebCore::FloatBoxExtent obscuredInsets();
+    WebCore::FloatBoxExtent unobscuredSafeAreaInsets();
+
+    double scale();
+    OptionSet<WebKit::ViewStabilityFlag> viewStability();
+    bool isFirstUpdateForNewViewSize();
+    bool allowShrinkToFit();
+    bool enclosedInScrollableAncestorView();
+    WebCore::VelocityData scrollVelocity();
+    WebKit::TransactionID lastLayerTreeTransactionID();
+};
+
+#endif

